### PR TITLE
[Procuration] Fix bug on state validation and add postal code logic

### PR DESCRIPTION
**Pull request by JeremieSamson**
This PR add an hide & show logic on the postal Code field.

1) When the user select FR in the country select

proxy_country_fr

2) When the user select something else than FR in the country select

proxy_country_non_fr

Same behavior with the vote country

1) When the user select FR in the vote country select

proxy_vote_country_fr

2) When the user select something else than FR in the vote country select

proxy_vote_country_non_fr

### CODE REVIEW

File: front/services/form/changeFieldsVisibility.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
+export default function (
+    countryFieldSelector,
+    postalCodeFieldSelector,
+    stateFieldSelector
+) {
+    const countryElement = dom(countryFieldSelector);
+    const postalCodeElement = dom(postalCodeFieldSelector);
+    const stateElement =dom(stateFieldSelector);
```
Comment (illusionOfParadise) [inline front/services/form/changeFieldsVisibility.js]: Space missing.

File: translations/validators.fr.yml [hunk 2]
```diff
@@ -211,6 +211,8 @@ procuration.vote_country.conditions.fr: Attention, vous ne pouvez être mandatai
 procuration.vote_country.conditions.other: Attention, vous ne pouvez être mandataire que pour trois procurations dont une seule maximum établie en France.
 procuration.vote_country.conditions.conditions: Merci de bien vouloir remplir la condition ci-dessous.
 procuration.state.not_empty: Vous devez remplir l'État/Province si le pays de résidence n'est pas en France.
+procuration.postal_code.not_empty: Vous devez remplir le code postal si le pays de résidence est en France.
+procuration.vote_postal_code.not_empty: Vous devez remplir le code postal de votre lieu de vote si le pays de résidence est en France.
```
Comment (illusionOfParadise) [inline translations/validators.fr.yml]:
```suggestion
procuration.vote_postal_code.not_empty: Vous devez remplir le code postal de votre lieu de vote si le pays de résidence est France.
```
And change in tests.

File: translations/validators.fr.yml [hunk 3]
```diff
@@ -211,6 +211,8 @@ procuration.vote_country.conditions.fr: Attention, vous ne pouvez être mandatai
 procuration.vote_country.conditions.other: Attention, vous ne pouvez être mandataire que pour trois procurations dont une seule maximum établie en France.
 procuration.vote_country.conditions.conditions: Merci de bien vouloir remplir la condition ci-dessous.
 procuration.state.not_empty: Vous devez remplir l'État/Province si le pays de résidence n'est pas en France.
+procuration.postal_code.not_empty: Vous devez remplir le code postal si le pays de résidence est en France.
```
Comment (illusionOfParadise) [inline translations/validators.fr.yml]:
```suggestion
procuration.postal_code.not_empty: Vous devez remplir le code postal si le pays de résidence est France.
```
And change in tests.

File: translations/validators.fr.yml (hunk 2 above)
Comment (JeremieSamson) [reply translations/validators.fr.yml]: Changed after a talk with our good Lord

**Pull request merged**